waku/waku_filter_v2/protocol.nim

```diff
@@ -176,20 +178,30 @@ proc pushToPeers(
   let msgHash =
     messagePush.pubsubTopic.computeMessageHash(messagePush.wakuMessage).to0xHex()
 
-  notice "pushing message to subscribed peers",
-    pubsubTopic = messagePush.pubsubTopic,
-    contentTopic = messagePush.wakuMessage.contentTopic,
-    target_peer_ids = targetPeerIds,
-    msg_hash = msgHash
+  wf.messageCache.expire(Moment.now())
+  if wf.messageCache.contains(msgHash):
```

Comment:
```suggestion
  if wf.messageCache.put(msgHash):
```

I think this is simpler and does exactly the same. It will call expire and will refresh timestamp also.